### PR TITLE
Feature/jlog

### PR DESCRIPTION
**Pull request by simeon-ned**
Reduce the size of following text, specifically do not use separate bullet for groups, just make list of them:"Thank you for providing that information. Based on the content you shared, I can summarize the key points:

This pull request adds analytical Jacobians for the logarithm (log) operation for all groups presented in `jaxlie`. These Jacobians are also known as right inverse Jacobians. They represent the partial derivative of the log operation with respect to the group element:

$$\text{Jlog}(T) = \frac{\partial \log_3(T)}{\partial T}$$

The implementation are based on derivations presented in the ["micro Lie theory" paper](https://arxiv.org/pdf/1812.01537), specifically equations 41c, 79, 126, 144, 163, and 179. Similar functionality is also implemented in established robotics libraries like Pinocchio, indicating the practical importance of these Jacobians in robotics applications.

### CODE REVIEW

File: jaxlie/_se3.py [hunk 1]
```diff
@@ -220,6 +213,52 @@ def adjoint(self) -> jax.Array:
...
+        st, ct = jnp.sin(theta), jnp.cos(theta)
+        inv_2_2ct = 1 / (2 * (1 - ct))
+
+        beta = jnp.where(theta < 1e-6,
+                         1 / 12 + t2 / 720,
+                         t2inv - st * tinv * inv_2_2ct)
+
+        beta_dot_over_theta = jnp.where(theta < 1e-6,
+                                        1 / 360,
+                                        -2 * t2inv * t2inv + (1 + st * tinv) * t2inv * inv_2_2ct)
```
Comment (brentyi) [inline jaxlie/_se3.py:244]: this looks like it can produce NaNs for reverse-mode autodiff, due to  https://jax.readthedocs.io/en/latest/faq.html#gradients-contain-nan-where-using-where

you can see how we fix it elsewhere: https://github.com/search?q=repo%3Abrentyi%2Fjaxlie%20shim&type=code

**Review by brentyi — COMMENTED** [review]
Thanks @simeon-ned! This is a great first pass.

I left some comments. It seems like the main things that need addressing are:
- Handling leading batch axes.
- Making sure we handle all divide by zero cases, including preventing revese-mode AD-specific NaNs[^1] for the `jlog` methods. 

For tests:
- Move tests to pytest.
- Tests for batch axes.
- Tests for autodiff through `jlog`. (perhaps using the `jacnumerical` or `_assert_jacobians_close` helpers, see `test_autodiff.py`)

Does that sound right? Am I missing anything?

[^1]: https://jax.readthedocs.io/en/latest/faq.html#gradients-contain-nan-where-using-where

File: jaxlie/_so2.py [hunk 2]
```diff
@@ -130,6 +130,16 @@ def normalize(self) -> SO2:
...
         )
 
+    @override
+    def jlog(self) -> jax.Array:
+        # Reference:
+        # For SO2 the jlog and right jacobians are trivially 1, 
+        # equation (126) from Micro-Lie theory:
+        # > https://arxiv.org/pdf/1812.01537
+
+        return jnp.array([1])
```
[the remark this answers is not in the document]
Comment (simeon-ned) [reply jaxlie/_so2.py:140]: sure

**Comment by brentyi** [comment]
Actually, for SE(2) and SE(3), are you able to reuse/refactor the code for the "V" and "V_inv" matrices?
https://github.com/brentyi/jaxlie/blob/84babf5c04ba6287cdec250ab1eed2f6d0a53988/jaxlie/_se2.py#L164-L175
https://github.com/brentyi/jaxlie/blob/84babf5c04ba6287cdec250ab1eed2f6d0a53988/jaxlie/_se3.py#L142-L159
https://github.com/brentyi/jaxlie/blob/84babf5c04ba6287cdec250ab1eed2f6d0a53988/jaxlie/_se2.py#L208-L221
https://github.com/brentyi/jaxlie/blob/84babf5c04ba6287cdec250ab1eed2f6d0a53988/jaxlie/_se3.py#L183-L205

These already seem to describe the parts of the left/right Jacobians that are the trickiest to compute.

**Comment by simeon-ned** [comment]
> Actually, for SE(2) and SE(3), are you able to reuse/refactor the code for the "V" and "V_inv" matrices?
> 
> https://github.com/brentyi/jaxlie/blob/84babf5c04ba6287cdec250ab1eed2f6d0a53988/jaxlie/_se2.py#L164-L175
> 
> 
> https://github.com/brentyi/jaxlie/blob/84babf5c04ba6287cdec250ab1eed2f6d0a53988/jaxlie/_se3.py#L142-L159
> 
> 
> https://github.com/brentyi/jaxlie/blob/84babf5c04ba6287cdec250ab1eed2f6d0a53988/jaxlie/_se2.py#L208-L221
> 
> 
> https://github.com/brentyi/jaxlie/blob/84babf5c04ba6287cdec250ab1eed2f6d0a53988/jaxlie/_se3.py#L183-L205
> 
> These already seem to describe the parts of the left/right Jacobians that are the trickiest to compute.

Great suggestion, should I encupsulate `V, V_inv` in to separate method or create  standalone function similar to `_skew` in SE3?

**Comment by brentyi** [comment]
I'd vote for a helper function for now so it's not exposed in the API, it's easier to add things to the API later than to remove them!

**Review by brentyi — APPROVED** [review]
I made a handful of changes, this looks ready to merge. Thanks for making this PR @simeon-ned!
